pkg/cmd/cli/cli.go

```diff
@@ -62,36 +62,40 @@ func NewCommandCLI(name, fullName string) *cobra.Command {
 	in := os.Stdin
 	out := os.Stdout
 
+	// OpenShift-specific commands
+	cmds.AddCommand(cmd.NewCmdBuildLogs(fullName, f, out))
+	cmds.AddCommand(cmd.NewCmdCancelBuild(fullName, f, out))
+	cmds.AddCommand(cmd.NewCmdConfig(fullName, "config"))
+	cmds.AddCommand(cmd.NewCmdDeploy(fullName, f, out))
+	cmds.AddCommand(cmd.NewCmdEdit(fullName, f, out))
+	cmds.AddCommand(cmd.NewCmdEnv(fullName, f, os.Stdin, out))
 	cmds.AddCommand(cmd.NewCmdLogin(fullName, f, in, out))
 	cmds.AddCommand(cmd.NewCmdLogout("logout", fullName+" logout", fullName+" login", f, in, out))
+	cmds.AddCommand(cmd.NewCmdNewApplication(fullName, f, out))
```

Comment:
The organization of the main help is going to be like "git help".  Ordered by function and utility.  The group boundaries will be based on the order.

----- Original Message -----
> >  	cmds.AddCommand(cmd.NewCmdLogin(fullName, f, in, out))
> >  	cmds.AddCommand(cmd.NewCmdLogout("logout", fullName+" logout", fullName+"
> >  	login", f, in, out))
> > +	cmds.AddCommand(cmd.NewCmdNewApplication(fullName, f, out))
> 
> hm, for what it's worth (not much i'm sure) i think alphabetic makes more
> sense.  if i'm trying to scan for the usage of a command i know the name of,
> it's easier to find if the list is alphabetical.  and if i don't know the
> name of the command i'm looking for, i still may have guesses which are
> going to lead me to navigate the list in an alphabetic search.
> 
> i don't really see the value in having the most common ones at the top, given
> that the whole list is going to be displayed anyway (in fact you could argue
> the most common ones should be at the bottom since that's what's going to be
> front and center after someone runs osc)
> 
> ----- Original Message -----
> > From: "Clayton Coleman" <notifications@github.com>
> > To: "openshift/origin" <origin@noreply.github.com>
> > Cc: "Ben Parees" <bparees@redhat.com>
> > Sent: Monday, May 18, 2015 1:34:59 PM
> > Subject: Re: [origin] Add support for 'osc stop|label' from upstream
> > (#1944)
> > 
> > >  	cmds.AddCommand(cmd.NewCmdLogin(fullName, f, in, out))
> > >  	cmds.AddCommand(cmd.NewCmdLogout("logout", fullName+" logout",
> > >  	fullName+"
> > >  	login", f, in, out))
> > > +	cmds.AddCommand(cmd.NewCmdNewApplication(fullName, f, out))
> > 
> > We don't want that.  The order is intentional, most used commands followed
> > by
> > lesser used commands.
> > 
> > ----- Original Message -----
> > > >  	cmds.AddCommand(cmd.NewCmdLogin(fullName, f, in, out))
> > > >  	cmds.AddCommand(cmd.NewCmdLogout("logout", fullName+" logout",
> > > >  	fullName+"
> > > >  	login", f, in, out))
> > > > +	cmds.AddCommand(cmd.NewCmdNewApplication(fullName, f, out))
> > > 
> > > I am not sure of the previous order but now the commands are
> > > **alphabetically** sorted in two groups: OpenShift-specific and
> > > Kubernetes
> > > wrappers.
> > > 
> > > ---
> > > Reply to this email directly or view it on GitHub:
> > > https://github.com/openshift/origin/pull/1944/files#r30515535
> > 
> > 
> > ---
> > Reply to this email directly or view it on GitHub:
> > https://github.com/openshift/origin/pull/1944/files#r30529266
> 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/1944/files#r30530444